packages/health-check/src/api/index.js

```diff
@@ -2,61 +2,75 @@ const { StatusCodes } = require("http-status-codes");
 const { sum, sumBy } = require("lodash");
 const db = require("../db");
 
-// getStatus returns the server's current health check status
-function getStatus() {
-  const disabled = db.get("disabled").value();
+/**
+ * Get status code that should be returned in the API response.
+ * - OK (200) in case everything is healthy
+ * - SERVICE_UNAVAILABLE (503) in case of any failures or if disabled
+ */
+function getStatusCode() {
+  // check whether the portal has been manually disabled
+  const disabled = getDisabled();
 
   if (disabled) {
     return StatusCodes.SERVICE_UNAVAILABLE;
   }
 
-  // Grab entry element from DB
-  const entry = db.get("entries").orderBy("date", "desc").head().value();
+  // grab one most recent critical entry element from DB
+  const entry = getCurrentCriticalEntry();
 
-  // Check that every critical check entry is up
-  if (entry && entry.checks.every(({ up, critical }) => up && critical)) {
+  // find out whether every check in the entry is up
+  if (entry && entry.checks.every(({ up }) => up)) {
     return StatusCodes.OK;
   }
 
-  // At least one check failed
+  // in case at least one check failed
   return StatusCodes.SERVICE_UNAVAILABLE;
```

Comment:
PP: I would typically flip these around and list all failure modes first and then return OK at the end. This allows you to focus on all the ways in which the negative outcome of the processing can be reached and then, having taken care of all the negative cases, you can safely return the positive one at the end.

```suggestion
// find out if any of the checks failed
if (!entry || entry.checks.any(({ up }) => !up)) {
  return StatusCodes.SERVICE_UNAVAILABLE;
}

// all checks passed!
return StatusCodes.OK;
```